pep.rst

```diff
@@ -1,13 +1,43 @@
+PEP: 9999
+Title: Tag Strings For Writing Domain-Specific Languages
+Author: Jim Baker <jim.baker@python.org>, Guido van Rossum <guido@python.org>, Paul Everitt <pauleveritt@me.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 01-Jun-2024
+Python-Version: 3.13
+
 Abstract
 ========
 
 This PEP introduces tag strings for custom, repeatable string processing. Tag strings
 are an extension to f-strings, with a custom function -- the "tag" -- in place of the
-`f` prefix. This function can then provide rich features such as safety checks, lazy
-evaluation, domain specific languages (DSLs) for web templating, and more.
+``f`` prefix. This function can then provide rich features such as safety checks, lazy
+evaluation, domain-specific languages (DSLs) for web templating, and more.
 
 Tag strings are similar to `JavaScript tagged templates <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates>`_
-and similar ideas in other languages.
+and related ideas in other languages.
+
+Tag functions can be very simple:
+
+.. code-block:: python
+
+    def greet(*args):
+        salutation = args[0].strip()
+        getvalue = args[1][0]
+        recipient = getvalue().upper()
+        return f"{salutation} {recipient}!"
+
+With this, ``greet`` can be used in a way that is familiar to f-strings:
+
+.. code-block:: python
+
+    name = "World"
+    greeting = greet"Hello {name}"
+    assert greeting == "Hello WORLD!"
+
+Below you can find richer examples. As a note, an implementation based on CPython 3.12
+exists, as discussed in this document.
```

Comment:
In general it's good to avoid language which says things are "simple" or "easy". This could turn away readers who don't immediately agree that it is indeed simple or easy.

I'd also probably suggest putting the usage first. I find that looking at the implementation and trying to deduce what it does is harder than working backwards from the usage in order to understand the implementation.

Lastly, in reading the example, there's a few subjective things I might change:

- I find that the indexing (e.g. `args[1][0]`) makes it a little hard to understand the structure of `args`. De-structuring is a bit more readable to my eyes.
- Modifying the `salutation` in a different way from the `recipient` could make it a littler easier to see where the parts of the original string literal have gone. I imagine that someone reading my suggested changes below would see that `hello` has been changed to `Hello` and that `world` has been changed to `WORLD` and would thus be primed to search the implementation for where these differences come from.

```suggestion
The example below shows how similar tag strings are to the ``f`` strings you're likely already familiar with. Albeit, with the ability to change how the literal string and
embedded values are interpreted:

.. code-block:: python

    name = "world"
    greeting = greet"hello {name}"
    assert greeting == "Hello WORLD!"

The function ``greet`` would be implemented thusly:

.. code-block:: python

    def greet(*args):
        salutation, recipient = args
        _, getvalue = recipient
        return f"{salutation.title()} {getvalue().upper()}!"

Below you can find richer examples. As a note, an implementation based on CPython 3.12
exists, as discussed in this document.
```

Reply:
Thanks for making the time @rmorshea ... I agree about both points.

Reply:
I changed a bit of the wording, just made a commit. Thanks @rmorshea

Reply:
After this change to the example it looks like some of those that follow need to be modified for consistency. I can try and make that contribution this evening unless someone else gets to it first.